pkg/chatbot/twitch_chatbot.go

```diff
@@ -0,0 +1,116 @@
+package chatbot
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gempir/go-twitch-irc/v2"
+	"github.com/nicklaw5/helix/v2"
+	"github.com/saxypandabear/twitchsongrequests/internal/constants"
+	"github.com/saxypandabear/twitchsongrequests/pkg/db"
+	"github.com/zmb3/spotify/v2"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+func SendChatMessage(broadcaster string, userStore db.UserStore, event *helix.EventSubChannelPointsCustomRewardRedemptionEvent) error {
+	u, err := userStore.GetUser(event.BroadcasterUserID)
+	if err != nil {
+		return err
+	}
+
+	chatClient := twitch.NewClient(broadcaster, "oauth:"+u.TwitchAccessToken)
+	chatClient.Join(broadcaster)
+
+	chatClient.OnConnect(func() {
+		spotifyToken := &oauth2.Token{
+			AccessToken:  u.SpotifyAccessToken,
+			RefreshToken: u.SpotifyRefreshToken,
+			Expiry:       *u.SpotifyExpiry,
+		}
+
+		spotifyConfig := &oauth2.Config{
+			ClientID:     os.Getenv(constants.SpotifyClientIDKey),
+			ClientSecret: os.Getenv(constants.SpotifyClientSecretKey),
+			Endpoint: oauth2.Endpoint{
+				AuthURL:  "https://accounts.spotify.com/authorize",
+				TokenURL: "https://accounts.spotify.com/api/token",
+			},
+		}
```

Comment:
The problem with this is that from what I recall, Spotify OAuth tokens expire really quickly. I created utility functions to try to handle this under the hood. Instead of creating the raw OAuth token and config, you could use the `util.FetchSpotifyToken()` and `util.RefreshSpotifyToke()` functions. Separately, you can/should use the existing utility function for creating the Spotify client `util.GetNewSpotifyClient()`

References:
---
https://github.com/SaxyPandaBear/TwitchSongRequests/blob/44b2f0ca1794d861a0c6793bea2bdc0b09cda590/pkg/api/callback.go#L126

https://github.com/SaxyPandaBear/TwitchSongRequests/blob/44b2f0ca1794d861a0c6793bea2bdc0b09cda590/pkg/api/callback.go#L133

https://github.com/SaxyPandaBear/TwitchSongRequests/blob/44b2f0ca1794d861a0c6793bea2bdc0b09cda590/pkg/api/callback.go#L156